test/xbgpu/test_correlation.py

```diff
@@ -144,9 +146,12 @@ def test_correlator(
     fill_random(rng, buf_visibilities_device, command_queue)
 
     # Calculate expected values
-    calculated_visibilities_host = np.zeros(buf_visibilities_host.shape, buf_visibilities_host.dtype)
+    calculated_visibilities_host = np.zeros(buf_visibilities_host.shape, np.int64)
```

Comment:
Why the move to `np.int64`? (from `np.int32`)

Reply:
Something I changed (probably just passing more spectra per heap in some circumstances) was causing int32 to overflow, so I upgraded the test to correctly saturate the same way the GPU code does (by first calculating in int64 then clamping).

Reply:
Noted and noted.